protos/flyteidl/admin/description_entity.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package flyteidl.admin;
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin";
+
+import "flyteidl/core/identifier.proto";
+import "flyteidl/admin/common.proto";
+
+// DescriptionEntity contains detailed description for the task/workflow/launch plan.
+// Documentation could provide insight into the algorithms, business use case, etc.
+message DescriptionEntity {
+  // One-liner overview of the entity.
+  string short_description = 1;
+  // Full user description with formatting preserved.
+  LongDescription long_description = 2;
+  // User-specified tags. These are arbitrary and can be used for searching
+  // filtering and discovering entities.
+  repeated string tags = 3;
+  // User-defined free-form key-value pair attributes. These are arbitrary
+  // and can be used for searching, filtering and discovering entities.
+  Labels labels = 4;
+  // Optional link to source code used to define this entity.
+  SourceCode source_code = 5;
+}
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
+  // long description - no more than 4KB
+  string values = 1;
+  // if the description sizes exceed some threshold we can offload the entire
+  // description proto altogether to an external data store, like S3 rather than store inline in the db
+  string uri = 2;
```

Comment:
```suggestion
  oneof content {
    // long description - no more than 4KB
    string value = 1;
    // if the description sizes exceed some threshold we can offload the entire
    // description proto altogether to an external data store, like S3 rather than store inline in the db
    string uri = 2;
  }
```